slo_generator/utils.py

```diff
@@ -60,7 +63,9 @@ def load_configs(path, ctx=os.environ, kind=None):
     return [cfg for cfg in configs if cfg]
 
 
-def load_config(path, ctx=os.environ, kind=None):
+def load_config(path: str,
+                ctx: os._Environ = os.environ,
+                kind: Optional[str] = None) -> Optional[dict]:
```

Comment:
`Optional[dict]` generates an error when `pytype` does its own linting and type checking (check [the logs](https://github.com/google/slo-generator/actions/runs/3303042130/jobs/5450547176) for more details):

```
File "/home/runner/work/slo-generator/slo-generator/slo_generator/api/main.py", line 101, in run_export: No attribute 'get' on None [attribute-error]
  In Optional[dict]
File "/home/runner/work/slo-generator/slo-generator/slo_generator/api/main.py", line 195, in process_batch_req: No attribute 'get' on None [attribute-error]
  In Optional[dict]
```

Replace with `dict`? It is probably the right type here, even if there is a possiblity to raise an exception.

Reply:
I don't think so, if we replace it with `dict` it will complains.

We need to check if it's not null before `get` anything from it, that's what I did before

Reply:
You are absolutely right, and that is why I like type hints! `mypy` and `pytype` are able to surface bugs like this one. I cloned your branch and ran a few tests. `config` can indeed be `None` and must be checked explicitly. Good catch!

For the sake of brevity and simplicity, can you surround the two faulty lines (101 and 195) with a `FIXME` (so we have a chance to fix this later) and temporarily disable the `pytype` warnings? I used this on lines 101 and 195 and was able to run `make pytype` successfully as a consequence:

```python
            # pytype: disable=attribute-error
            # FIXME `load_config()` returns `Optional[dict]` so `config` can be `None`
            exporter_conf = config.get('pubsub_batch_handler')
            # pytype: enable=attribute-error
```

Do this on both lines and we should be good to go.

Reply:
I did this but now `pylint` fails because of `FIXME`